lib/spack/llnl/util/tty/log.py

```diff
@@ -765,7 +765,13 @@ def _writer_daemon(stdin_multiprocess_fd, read_multiprocess_fd, write_fd, echo,
 
                 if in_pipe in rlist:
                     # Handle output from the calling process.
-                    line = _retry(in_pipe.readline)()
+                    try:
+                        line = _retry(in_pipe.readline)()
+                    # Sometimes build processes produce invalid unicode that 
+                    # the decoder can't handle. Ignore it and move on.
+                    except UnicodeDecodeError:
+                        continue
```

Comment:
We should log the error here -- if we're going to lose part of the output, we should at least let the user know that we lost it.

Reply:
This has been updated by Paul accordingly as far as I saw during rebase to current develop (it's one of the commits)